### DISCUSSION TITLE
Battery icon missing in latest Meslo LG S DZ Nerd Font Italic

### DISCUSSION THREAD

**Discussion by heafnerj** [ryanoasis/nerd-fonts#1463] [Q&A]
I want to update to the latest version of the Meslo fonts it use width Oh-My-Zsh and the p10k theme. When I did everything seemed to work perfectly except that the battery icon in my Terminal prompt was not displaying. The current font file is "Meslo LG S DZ Italic Nerd Font Complete.ttf" and the corresponding new one is "MesloLGSDZNerdFontItalic.ttf" as far as I can tell. Will running the theme's setup script restore the missing battery icon or is something else going on?

**Comment by Finii** [comment]
Sorry to hear of your difficulties.
I suspect that the battery icon you use is one of the icons that moved codepoints with `Nerd Fonts 3.x.x`.
Is your old-and-working font file a pre `3.x.x` version?
~~If it is `3.0.0` or later,~~ Please copy and paste the icon here as text, so that we can determine the codepoint.
Of copy just the one missing glyph and paste it into the cheat sheet:

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/2abe5cde-9452-428c-aed8-0d715fd6262b)

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/70a66d69-a123-405c-9089-b0c04a774c18)

**Reply by heafnerj** [reply]
According to the Cheat Sheet the icon () is nf_mdi_battery_90 code f581 and indeed has been removed. Yes, the old-and-working font is very much pre 3.x.x. macOS Get Info reports version 1.210. I would like to use eza (https://github.com/eza-community/eza) (forked from https://github.com/ogham/exa).

**Reply by Finii** [reply]
For `nf-mdi-battery_90` (`F581`) you can use `nf-md-battery_90` (`F00821`).
I assume this is in your settings files.
Some (at least one) prompt project added a settings translator function. Maybe you have that.
The generic [`nerdfix`](https://github.com/loichyan/nerdfix) can be used otherwise - that tool will fix the codepoints in your settings.

I must admit I have no detail knowledge here, as I personally do not use any sophisticated prompt stuff etc.

`eza` seems to use (in theiy demo) a Nerd Font 3.00+ font, so codepoint-wise you should be fine.
Also browsing through https://github.com/eza-community/eza/blob/main/src/output/icons.rs looks like Nerd Font 3.0.0+.

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/5482a42b-35a8-419e-a157-0be72422f99d)

_Font name pattern `...SansM` is new with Nerd Font 3.0.0, before it kept the original `...SansMono`_

**Reply by heafnerj** [reply]
Thank you very much. Let me dig into this.

**Reply by heafnerj** [reply]
I found the code points in my '.p10k.zsh' file and replacements for the removed icons in the Cheat Sheet. I think I can just manually replace them.

**Reply by heafnerj** [reply]
The problem is solved. I updated to the latest [Powerlevel10k](https://github.com/romkatv/powerlevel10k) theme and it supports the newest Nerd Fonts. Everything works perfectly now, including [eza](https://github.com/eza-community/eza). Thank you for your help.
